server/fileloader.mjs

```diff
@@ -25,7 +27,12 @@ async function downloadLink(link) {
     };
   }
 
-  const response = await fetch(link, requestEtag ? { headers: { 'If-None-Match': requestEtag } } : {});
+  // if the link is to our own server, use the local link instead so we can skip external auth, if any
+  var actual_link = link;
+  if (link.startsWith(config.EXTERNAL_ADDRESS)) {
```

Comment:
This works when it's `null`?

Reply:
Yes, it does. It doesn't work when it's empty but works when it's null. I can make it more explicit to avoid relying on javascript's un-intuitive behavior.